pkg/detectors/openai/openai.go

```diff
@@ -103,6 +70,67 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 	return
 }
 
+func verifyToken(ctx context.Context, client *http.Client, token string) (bool, map[string]string, error) {
+	// Undocumented API
+	// https://api.openai.com/v1/organizations
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.openai.com/v1/organizations", nil)
+	if err != nil {
+		return false, nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	res, err := client.Do(req)
+	if err != nil {
+		return false, nil, err
+	}
+	defer func() {
+		_, _ = io.Copy(io.Discard, res.Body)
+		_ = res.Body.Close()
+	}()
+
+	switch res.StatusCode {
+	case 200:
```

Comment:
@rgmz previously verification was based on range of 2xx Http status. Is there any particular reason to change it ?

Reply:
Many of the detectors are based on a template with generic conditions. I think it's prudent to only consider something verified if it matches known/documented status codes.

Unexpected API/site changes, and thus status code changes, will propagate as an error rather than creating false-positives.

Reply:
The API reference of OpenAI has well defined the [error codes](https://platform.openai.com/docs/guides/error-codes) (also In there [python library](https://github.com/openai/openai-python/blob/main/src/openai/_client.py)). In my opinion, Its good to parse the response but It wont make any effect on verification.

Reply:
> In my opinion, Its good to parse the response but It wont make any effect on verification.

I agree that parsing error codes doesn't make sense for OpenAI (although there are some APIs where 401 vs 403 is an importance difference).

I don't think it's a good practice to be liberal with what status codes are considered "valid" when only a specific one is expected. It's a recipe for false positives, imo.